src/Detector.php

```diff
@@ -64,6 +64,24 @@ public function determineMimeType(\SplFileObject $file): string
         return 'text/plain; charset=utf-8';
     }
 
+    /**
+     * Read width and height from a image file.
+     *
+     * @param string $filePath
+     * @return array [$width, $height]
+     * @throws \InvalidArgumentException Unreadable SplFileInfo
+     */
+    public function determineDimensions(string $filePath): array
+    {
+        $stream = new \SplFileInfo($filePath);
```

Comment:
Hi @peter279k,

Would you move this outside the function? So that `determineDimensions` _accepts_ an `SplFileInfo` rather than a file path string?

Sorry if there has been some misunderstanding; I don't want the API based around filepaths, instead something more abstract as that provides more long-term capabilities.